### PR TITLE
Upgrade global styles

### PR DESCRIPTION
**Pull request by zchsh**
:tickets: [Asana ticket](https://app.asana.com/0/1100423001970639/1199185162145231/f)
:mag: [Deploy preview](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/)

This PR upgrades to the latest `@hashicorp/mtkg-global-styles` release. This release cuts down on the amount of global CSS we ship as part of that package. This PR also upgrades to new versions of related components, in particular `@hashicorp/react-content`, which is used on all docs pages.

### Single page views

- [/](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/) vs [live](https://www.boundaryproject.io/)
- [/home](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/home) vs [live](https://www.boundaryproject.io/home)
  - note: duplicate of `/`, we have [a ticket to address this](https://app.asana.com/0/1100423001970639/1199638081194663/f)
- [/404](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/404) vs [live](https://www.boundaryproject.io/404)
- [/community](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/community) vs [live](https://www.boundaryproject.io/community)
- [/downloads](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/downloads) vs [live](https://www.boundaryproject.io/downloads)
- [/security](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/security) vs [live](https://www.boundaryproject.io/security)

### Docs views

All views that use `docs-page` use an updated `content` component with localized `table` styles, as those styles are no longer shipped as global CSS. The updated `docs-page` component also contains minor alignment fixes.

- [/api-docs/[[...page]]](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/api-docs/) vs [live](https://www.boundaryproject.io/api-docs/)
- [/docs/[[...page]]](https://boundary-git-fork-zchsh-zsupgrade-global-styles-hashicorp.vercel.app/docs/) vs [live](https://www.boundaryproject.io/docs/)

### CODE REVIEW

**Comment by jescalan** [comment]
Removing myself as a reviewer since this is a WIP draft - @malnick I think Zach will tag us for review when it's ready!

File: website/components/footer/index.jsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Link from 'next/link'
 export default function Footer({ openConsentManager }) {
   return (
     <footer className="g-footer">
-      <div className="g-container">
+      <div className="g-grid-container">
```
Comment (zchsh) [inline website/components/footer/index.jsx:6]: Have replaced all `g-container` with `g-grid-container`.

File: website/components/merch-desktop-client/merch-desktop-client.module.css [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .container {
   width: 100%;
-  border: 1px solid var(--gray-6);
+  border: 1px solid var(--gray-5);
```
Comment (zchsh) [inline website/components/merch-desktop-client/merch-desktop-client.module.css:3]: Fixes an issue where border colour was inconsistent.

File: website/components/openapi-page/index.jsx [hunk 3]
```diff
@@ -54,7 +54,9 @@ function OpenApiPage({
               <h1 className={`${styles.categoryHeading} g-type-display-4`}>
                 {operationCategory.name}
               </h1>
-              <div ref={operationsRef}>
+              {/* @TODO: remove g-type-body-long className once https://github.com/hashicorp/react-components/pull/280 lands,
+              and has been pulled into this project. Until then, this className is needed to correct a typo in the g-content component. */}
+              <div className="g-type-long-body" ref={operationsRef}>
```
Comment (zchsh) [inline website/components/openapi-page/index.jsx:59]: Note: this is a really annoying thing I wish I caught earlier. However, it has very limited effect, as on all other docs pages (apart from these generated OpenAPI ones), we have our typography plugin in place, so correcting the className issue isn't critical (as most if not all content appears with the right type style anyways).

I have a fix in https://github.com/hashicorp/react-components/pull/280, which I'm going to merge tomorrow AM, and will likely be able to resolve this issue and close this out before merging this PR.

File: website/content/security.mdx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+---
```
Comment (zchsh) [inline website/content/security.mdx:1]: Have switched over to the `markdown-page` approach to `/security` to match other docs properties.

File: website/pages/_app.js [hunk 5]
```diff
@@ -38,7 +38,7 @@ export default function App({ Component, pageProps }) {
         <meta name="og:description" property="og:title" content={description} />
       </HashiHead>
       {ALERT_BANNER_ACTIVE && (
-        <AlertBanner {...alertBannerData} product="boundary" />
+        <AlertBanner {...alertBannerData} product="boundary" hideOnMobile />
```
Comment (zchsh) [inline website/pages/_app.js:41]: Hides alert-banner on mobile, to align with versioned docs needs. This matches previous behaviour - which was previously enabled by globally scoped overrides in `docs-page`.

File: website/pages/community/index.jsx [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import VerticalTextBlockList from '@hashicorp/react-vertical-text-block-list'
 import SectionHeader from '@hashicorp/react-section-header'
 import Head from 'next/head'
+import s from './style.module.css'
 
 export default function CommunityPage() {
   return (
-    <div id="p-community">
+    <div className={s.root}>
```
Comment (zchsh) [inline website/pages/community/index.jsx:8]: Removes the deprecated `--site-max-width` custom property, and refactors to CSS modules.

File: website/components/openapi-page/index.jsx (hunk 3 above)
Comment (zchsh) [reply website/components/openapi-page/index.jsx]: Fixed now 😅

File: website/components/openapi-page/index.jsx [hunk 7]
```diff
@@ -79,7 +79,7 @@ function OpenApiPage({
               <h1 className={`${styles.pageHeading} g-type-display-2`}>
                 {info.title}
               </h1>
-              <p className={`${styles.landingPlaceholder} g-type-body-long`}>
+              <p className={`${styles.landingPlaceholder} g-type-long-body`}>
```
Comment (zchsh) [inline website/components/openapi-page/index.jsx:82]: 🤦 why are words and names so hard

**Review by BRKalow — APPROVED** [review]
✨

**Pull request merged**